client/my-sites/hosting/github/github-connect-card/index.tsx

```diff
@@ -1,16 +1,106 @@
+import FormFieldset from 'calypso/components/forms/form-fieldset';
+import FormSelect from 'calypso/components/forms/form-select';
+import FormTextInput from 'calypso/components/forms/form-text-input';
+import SocialLogo from 'calypso/components/social-logo';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import { RepoBranch, useGithubBranches } from './use-github-branches';
+import { Repo, useGithubRepos } from './use-github-repos';
+
+import './style.scss';
 
-// todo - just a placeholder for now as the implementation of this
-// component falls under another PR.
 export const GithubConnectCard = () => {
-	const { __ } = useI18n();
+	const siteId = useSelector( getSelectedSiteId );
+
+	const [ selectedRepo, setSelectedRepo ] = useState< Repo >();
+	const [ selectedBranch, setSelectedBranch ] = useState< RepoBranch >();
+	const { data: repos, isLoading: isLoadingRepos } = useGithubRepos( siteId );
+	const { data: branches, isLoading: isLoadingBranches } = useGithubBranches(
+		siteId,
+		selectedRepo?.name
+	);
+
+	useEffect( () => {
+		if ( repos ) {
+			setSelectedRepo( selectedRepo || repos[ 0 ] );
+		}
+		if ( branches ) {
+			setSelectedBranch( branches[ 0 ] );
+		}
+	}, [ repos, branches, selectedRepo ] );
+
+	const showSpinner = ! repos && ! branches && ( isLoadingRepos || isLoadingBranches );
+
+	const handleRepoSelect = ( repoName: string ) => {
+		const selectedRepo = repos?.find( ( repo ) => repo.full_name === repoName );
+		setSelectedRepo( selectedRepo );
+	};
+
+	const handleBranchSelect = ( branchName: string ) => {
+		const selectedBranch = branches?.find( ( branch ) => branch.name === branchName );
+		setSelectedBranch( selectedBranch );
+	};
+
 	return (
-		<Card className="github-hosting-card">
-			<img className="github-hosting-icon" src={ iconGitHub } alt="" />
+		<Card className="connect-branch-card">
+			<SocialLogo className="material-icon" icon="github" size={ 32 } />
 			<CardHeading>{ __( 'Connect Branch' ) }</CardHeading>
+			<div>
+				<p>{ __( 'Changes pushed to connected branch will be deployed automatically' ) }</p>
+				{ ! showSpinner ? (
+					<FormFieldset className="connect-branch__fields">
+						<FormSelect
+							id="repositories"
+							className="connect-branch__field"
+							onChange={ ( event ) => handleRepoSelect( event.currentTarget.value ) }
+							value={ selectedRepo?.full_name }
+						>
+							{ repos?.map( ( repo ) => (
+								<option value={ repo.full_name } key={ repo.name }>
+									{ repo.full_name }
+								</option>
+							) ) }
+						</FormSelect>
+						<FormSelect
+							id="branches"
+							className="connect-branch__field"
+							onChange={ ( event ) => handleBranchSelect( event.currentTarget.value ) }
+							value={ selectedBranch?.name }
+						>
+							{ branches?.map( ( branch ) => (
+								<option value={ branch.name } key={ branch.name }>
+									{ branch.name }
+								</option>
+							) ) }
+						</FormSelect>
+						<Button
```

Comment:
Can we connect this up so that it calls `/sites/{site_id}/hosting/github/connect?repo=...&branch=...&base_path=....`

The `base_path` will be optional but the others need to be set before this button can be clicked.